pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -145,11 +147,37 @@ func (r *ReconcileNMState) applyRBAC(instance *nmstatev1beta1.NMState) error {
 
 func (r *ReconcileNMState) applyHandler(instance *nmstatev1beta1.NMState) error {
 	data := render.MakeRenderData()
+	// Register ToYaml template method
+	data.Funcs["toYaml"] = nmstaterenderer.ToYaml
+	// Prepare defaults
+	masterExistsNoScheduleToleration := corev1.Toleration{
+		Key:      "node-role.kubernetes.io/master",
+		Operator: corev1.TolerationOpExists,
+		Effect:   corev1.TaintEffectNoSchedule,
+	}
+	amd64ArchOnMasterNodeSelector := map[string]string{
+		"beta.kubernetes.io/arch":        "amd64",
+		"node-role.kubernetes.io/master": "",
+	}
+	amd64AndCRNodeSelector := map[string]string{
+		"beta.kubernetes.io/arch": "amd64",
+	}
+	if instance.Spec.NodeSelector != nil {
+		for k, v := range instance.Spec.NodeSelector {
+			amd64AndCRNodeSelector[k] = v
+		}
+	}
```

Comment:
We can remove the for statement:

```golang
amd64AndCRNodeSelector = instance.Spec.NodeSelector
if amd64AndCRNodeSelector == nil {
    amd64AndCRNodeSelector = map[string]string{}
}
amd64AndCRNodeSelector["beta.kubernetes.io/arch"] = "amd64
```

Reply:
btw, maybe we want to have the same at CNAO, so we don't need to have "beta.kubevirt.io/arg" always at NetworkAddonsConfig since container only works at "amd64".

Reply:
Yes, we have this node selector added as default in CNAO
https://github.com/kubevirt/cluster-network-addons-operator/blob/master/pkg/network/placement_configuration.go